gnome-initial-setup/pages/site/gis-site-search-entry.c

```diff
@@ -0,0 +1,455 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+
+/* Copyright (C) 2008 Red Hat, Inc.
+ * Copyright (C) 2018 Endless Mobile, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see <http://www.gnu.org/licenses/>.
+ *
+ * Written by:
+ *     Travis Reitter <travis.reitter@endlessm.com>
+ */
+
+#include <string.h>
+#include <gio/gio.h>
+#include <glib/gi18n.h>
+#include <gtk/gtk.h>
+
+#include "gis-site-search-entry.h"
+
+/**
+ * SECTION:GisSiteSearchEntry
+ * @Title: GisSiteSearchEntry
+ *
+ * A subclass of #GtkSearchEntry that provides autocompletion on
+ * #GisSite<!-- -->s
+ */
+
+typedef struct {
+  GisSite          *site;
+  GPtrArray        *sites;
+  GtkTreeModel     *model;
```

Comment:
I just realized that you can remove this stored reference from here; the entry completion already holds a reference to the model, so you can get it from there when needed. Also you can remove the `GtkTreeModel` argument from `set_site_internal()`